cl/lib/tests.py

```diff
@@ -219,6 +232,67 @@ def test_making_docket_number_core(self) -> None:
         # an empty string.
         self.assertEqual(make_docket_number_core(None), "")
 
+    def test_avoid_generating_docket_number_core(self) -> None:
+        """Can we avoid generating docket_number_core when the docket number
+        format doesn't match a valid format or if a string contains more than
+        one docket number?
+        """
+
+        # Not valid docket number formats for district, bankruptcy or appellate
+        self.assertEqual(make_docket_number_core("Nos. C 123-80-123-82"), "")
+        self.assertEqual(make_docket_number_core("Nos. C 123-80-123"), "")
+        self.assertEqual(
+            make_docket_number_core("Nos. 212-213, Dockets 27264, 27265"), ""
+        )
+
+        # Multiple valid docket numbers
+        self.assertEqual(
+            make_docket_number_core(
+                "Nos. 14-13542, 14-13657, 15-10967, 15-11166"
+            ),
+            "",
+        )
+        self.assertEqual(make_docket_number_core("12-33112, 12-33112"), "")
+        self.assertEqual(
+            make_docket_number_core(
+                "CIVIL ACTION NO. 7:17-CV-00426,  7:17-CV-00426"
+            ),
+            "",
+        )
+
+    def test_clean_docket_number(self) -> None:
+        """Can we clean and return a docket number if it has a valid format?"""
+
+        # Not valid docket number formats for district, bankruptcy or appellate
+        # not docket number returned
+        self.assertEqual(clean_docket_number("Nos. C 123-80-123-82"), "")
+        self.assertEqual(clean_docket_number("Nos. C 123-80-123"), "")
+        self.assertEqual(clean_docket_number("Nos. 212-213"), "")
+
+        # Multiple valid docket numbers, not docket number returned
+        self.assertEqual(
+            clean_docket_number("Nos. 14-13542, 14-13657, 15-10967, 15-11166"),
+            "",
+        )
+        self.assertEqual(clean_docket_number("12-33112, 12-33112"), "")
+
+        # One valid docket number, return the cleaned number
+        self.assertEqual(
+            clean_docket_number("CIVIL ACTION NO. 7:17-CV-00426"),
+            "7:17-CV-00426",
+        )
+        self.assertEqual(
+            clean_docket_number("Case No.1:19-CV-00118-MRB"), "1:19-CV-00118"
+        )
+        self.assertEqual(clean_docket_number("Case 12-33112"), "12-33112")
+        self.assertEqual(clean_docket_number("12-33112"), "12-33112")
+        self.assertEqual(
+            clean_docket_number("12-cv-01032-JKG-MJL"), "12-cv-01032"
+        )
+        self.assertEqual(
+            clean_docket_number("Nos. 12-213, Dockets 27264, 27265"), "12-213"
+        )
```

Comment:
I think this is OK, but it worries me a bit. We'll see about this one, I think.

Reply:
I confirmed this one, this is from an opinion that has the following docket number:
"Nos. 212-213, Dockets 27264, 27265"
https://ia803106.us.archive.org/9/items/law.free.cap.f2d.297/32.226451.json

Actually the `212-213` is not a valid `docket_number`, so I added a test case where you can see the method returns "".

This docket was overwritten because without the `clean_docket_number` method, it generated the following `docket_number_core` `12000213` that belongs to the case `12-213` the one that now appears in the docket: https://www.courtlistener.com/docket/64009291/united-states-v-oppedisano/

With the fix no `docket_number_core` will be generated for cases like these.

I added the test above:
`self.assertEqual( clean_docket_number("Nos. 12-213, Dockets 27264, 27265"), "12-213")`
Just to simulate a situtation where the opinion has only one valid docket number in this case `12-213`,  so that the docket valid number is returned. If there are more than one valid docket number like `"Nos. 14-13542, 14-13657, 15-10967, 15-11166"` no `docket_number` is returned and no `docket_number_core` is generated.